spec.bs

```diff
@@ -386,49 +396,91 @@ Each {{URLPattern}} object has an associated <dfn for=URLPattern>hash component<
 <div algorithm>
   To <dfn>canonicalize protocol</dfn> given a string |value| and a string |type|:
 
-  1. TODO |value| |type|
+  1. Let |strippedValue| be the given |value| with a single trailing U+003A (:) removed, if any.
+  1. If |type| is "pattern" then return |strippedValue|.
+  1. Let |dummyURL| be a new [=url=] record.
```

Comment:
```suggestion
  1. Let |dummyURL| be a new [=URL record=].
```

here and elsewhere

Reply:
It seems like the algorithms are OK with an empty URL record... a bit surprising, but it seems to work. You could double-check with whatwg-url if you wanted.